src/NLog/Targets/EventLogTarget.cs

```diff
@@ -270,11 +271,20 @@ protected override void Write(LogEventInfo logEvent)
 
             EventLogEntryType entryType = GetEntryType(logEvent);
 
-            int eventId = EventId.RenderInt(logEvent, 0, "EventLogTarget.EventId");
+            string renderEventId = RenderLogEvent(EventId, logEvent);
+            string renderCategory = RenderLogEvent(Category, logEvent);
 
-            short category = Category.RenderShort(logEvent, 0, "EventLogTarget.Category");
+            int eventId = 0;
+            if (!string.IsNullOrEmpty(renderEventId) && !int.TryParse(renderEventId, NumberStyles.Integer, CultureInfo.InvariantCulture, out eventId))
+            {
+                InternalLogger.Warn("{0} WriteEntry failed to parse EventId={1}", this, renderEventId);
+            }
 
-          
+            short category = 0;
+            if (!string.IsNullOrEmpty(renderCategory) && !short.TryParse(renderCategory, NumberStyles.Integer, CultureInfo.InvariantCulture, out category))
```

Comment:
why did we remove the `RenderShort` and `RenderInt?` Because there aren't public?

Reply:
Yes not public

Reply:
Proposal: or make public, or duplicate the code into this package :)

It's easier to maintain then

please share your thoughts

Reply:
I was thinking everything would be changed, when NLog 6.0 comes and explodes everything into smaller pieces.

Reply:
yep, and it's easier to structural fix it with NLog 6 if we copy-paste now the methods, agree?

Reply:
Have now replace internal LayoutHelpers with public ConversionHelpers.